starport/interface/cli/starport/cmd/network_chain_show.go

```diff
@@ -28,25 +39,62 @@ func networkChainShowHandler(cmd *cobra.Command, args []string) error {
 
 	chainID := args[0]
 
-	c, err := nb.ShowChain(context.Background(), chainID)
+	// Get flags
+	home, _, err := getHomeFlags(cmd)
 	if err != nil {
 		return err
 	}
-	chainyaml, err := yaml.Marshal(c)
+	showGenesis, err := cmd.Flags().GetBool(genesisFlag)
 	if err != nil {
 		return err
 	}
-
-	info, err := nb.LaunchInformation(context.Background(), chainID)
+	showPeers, err := cmd.Flags().GetBool(peersFlag)
 	if err != nil {
 		return err
 	}
-	infoyaml, err := yaml.Marshal(info)
+	if showGenesis && showPeers {
+		return fmt.Errorf("%s and %s flags cannot be set both", genesisFlag, peersFlag)
+	}
+
+	// Fetch launch information
+	info, err := nb.LaunchInformation(context.Background(), chainID)
 	if err != nil {
 		return err
 	}
 
-	fmt.Printf("\nChain:\n---\n%s\n\nLaunch Information:\n---\n%s", string(chainyaml), string(infoyaml))
+	switch {
+	case showGenesis:
+		// Generate the genesis in a temporary directory and show the content
+		tmpHome, err := nb.GenerateTemporaryGenesis(cmd.Context(), chainID, home, info)
```

Comment:
```suggestion
		home, err := nb.SimulateConfigs(cmd.Context(), chainID)
```

where `home`s type is is:
```go
type chainHome string

// to avoid manually building path string in the code blow
// `genesisPath := fmt.Sprintf("%s/config/genesis.json", tmpHome)`
func (h chainHome) GenesisPath() string
```

?

Reply:
Yes it makes sense to move `config/genesis.json` out of cmd. Not sure about `SimulateConfigs` there is no simulation in this function

Reply:
I felt like this kind of a simulation of a home dir -with a genesis and other sample configs-, in terms of how it would look like in reality without actually starting the chain. (not sdk simulations or anything)

Reply:
"simulation" here refers to trying to start the chain and see if there is any error